adapter/pkg/operator/controllers/dp/api_controller.go

```diff
@@ -900,22 +912,26 @@ func addIndexes(ctx context.Context, mgr manager.Manager) error {
 		func(rawObj k8client.Object) []string {
 			api := rawObj.(*dpv1alpha1.API)
 			var httpRoutes []string
-			for _, ref := range api.Spec.ProdHTTPRouteRefs {
-				if ref != "" {
-					httpRoutes = append(httpRoutes,
-						types.NamespacedName{
-							Namespace: api.Namespace,
-							Name:      ref,
-						}.String())
+			if api.Spec.Production != nil {
+				for _, ref := range api.Spec.Production[0].HTTPRouteRefs {
+					if ref != "" {
+						httpRoutes = append(httpRoutes,
+							types.NamespacedName{
+								Namespace: api.Namespace,
+								Name:      ref,
+							}.String())
+					}
 				}
 			}
-			for _, ref := range api.Spec.SandHTTPRouteRefs {
-				if ref != "" {
-					httpRoutes = append(httpRoutes,
-						types.NamespacedName{
-							Namespace: api.Namespace,
-							Name:      ref,
-						}.String())
+			if api.Spec.Sandbox != nil {
+				for _, ref := range api.Spec.Sandbox[0].HTTPRouteRefs {
+					if ref != "" {
```

Comment:
this is already validated in api webhook(isEmptyStringsInArray)

Reply:
This check need to add for resolve array index out exception